src/IO/MySQLBinlogEventReadBuffer.cpp

```diff
@@ -4,9 +4,12 @@
 namespace DB
 {
 
-MySQLBinlogEventReadBuffer::MySQLBinlogEventReadBuffer(ReadBuffer & in_)
-    : ReadBuffer(nullptr, 0, 0), in(in_)
+MySQLBinlogEventReadBuffer::MySQLBinlogEventReadBuffer(ReadBuffer & in_, size_t checksum_signature_length_)
+    : ReadBuffer(nullptr, 0, 0), in(in_), checksum_signature_length(checksum_signature_length_)
 {
+    if (checksum_signature_length)
+        checksum_buf = new char[checksum_signature_length];
+
     nextIfAtEnd();
```

Comment:
`nextIfAtEnd()` may throw an exception, it will lead to memory leak. Since size of checksum is small and limited, we can just rename `CHECKSUM_CRC32_SIGNATURE_LENGTH` to `MAX_CHECKSUM_SIGNATURE_LENGTH` and use `char checksum_buf[MAX_CHECKSUM_SIGNATURE_LENGTH];` as before, without dynamic allocations. If `checksum_signature_length_` is greater then buffer size, throw `LOGICAL_ERROR`.

Reply:
Done.